DifferentiationInterface/ext/DifferentiationInterfaceFiniteDiffExt/onearg.jl

```diff
@@ -56,87 +78,118 @@ function DI.prepare_derivative(
         df = similar(y)
         cache = GradientCache(df, x, fdtype(backend), eltype(y), FUNCTION_NOT_INPLACE)
     end
-    return FiniteDiffOneArgDerivativePrep(cache)
+    relstep = if isnothing(backend.relstep)
+        default_relstep(fdtype(backend), eltype(x))
+    else
+        backend.relstep
+    end
+    absstep = isnothing(backend.absstep) ? relstep : backend.relstep
+    return FiniteDiffOneArgDerivativePrep(cache, relstep, absstep)
 end
 
 ### Scalar to scalar
 
 function DI.derivative(
     f,
-    ::FiniteDiffOneArgDerivativePrep{Nothing},
+    ::FiniteDiffOneArgDerivativePrep{Nothing,Nothing,Nothing},
     backend::AutoFiniteDiff,
     x,
     contexts::Vararg{DI.Context,C},
 ) where {C}
     fc = DI.with_contexts(f, contexts...)
-    return finite_difference_derivative(fc, x, fdtype(backend))
+    relstep = if isnothing(backend.relstep)
+        default_relstep(fdtype(backend), eltype(x))
+    else
+        backend.relstep
+    end
+    absstep = isnothing(backend.absstep) ? relstep : backend.relstep
+    return finite_difference_derivative(fc, x, fdtype(backend); relstep, absstep)
 end
 
 function DI.value_and_derivative(
     f,
-    ::FiniteDiffOneArgDerivativePrep{Nothing},
+    ::FiniteDiffOneArgDerivativePrep{Nothing,Nothing,Nothing},
     backend::AutoFiniteDiff,
     x,
     contexts::Vararg{DI.Context,C},
 ) where {C}
     fc = DI.with_contexts(f, contexts...)
     y = fc(x)
-    return y, finite_difference_derivative(fc, x, fdtype(backend), eltype(y), y)
+    relstep = if isnothing(backend.relstep)
+        default_relstep(fdtype(backend), eltype(x))
+    else
+        backend.relstep
+    end
+    absstep = isnothing(backend.absstep) ? relstep : backend.relstep
+    return y,
+    finite_difference_derivative(fc, x, fdtype(backend), eltype(y), y; relstep, absstep)
 end
 
 ### Scalar to array
 
 function DI.derivative(
     f,
-    prep::FiniteDiffOneArgDerivativePrep{<:GradientCache},
+    prep::FiniteDiffOneArgDerivativePrep{<:GradientCache,<:Number,<:Number},
```

Comment:
again this is not necessary because you already make sure they are numbers ;)
```suggestion
    prep::FiniteDiffOneArgDerivativePrep{<:GradientCache},
```